tests/Doctrine/Tests/Mocks/DatabasePlatformMock.php

```diff
@@ -20,18 +20,22 @@ class DatabasePlatformMock extends AbstractPlatform
     /** @var bool */
     private $_prefersSequences = false;
 
-    /**
-     * {@inheritdoc}
-     */
-    public function prefersIdentityColumns()
+    public function prefersIdentityColumns(): bool
```

Comment:
While this approach is okay, it would be better if these mocks were reworked using the PHPUnit mocking API. See https://github.com/doctrine/dbal/pull/3555 and https://github.com/doctrine/dbal/pull/3474 for example. This could be done separately but before handling other API changes.

Reply:
See https://github.com/doctrine/orm/issues/8886#issuecomment-893391508

I'm planning to revise the mocks next weekend. This is a major pain point of the DBAL upgrade in this repository: too much mocking of classes that belong to a different package.

Reply:
Awesome!